python/sglang/srt/layers/fused_moe/layer.py

```diff
@@ -572,6 +588,18 @@ def process_weights_after_loading(self, layer: Module) -> None:
                     start += shard_size
 
             layer.w13_scale = torch.nn.Parameter(max_w13_scales, requires_grad=False)
+            # If ROCm, apply weight padding (min. Mem channel contention) only if set
+            if is_hip() and bool(int(os.getenv("MOE_PADDING", "0"))):
```

Comment:
move all `is_hip` under a single branch.
e.g., L555

Reply:
@merrymercy understand, the order of data crunching makes me intend to keep padding at very last to avoid error prone situation from intervening `normalize_`, `_dequantize`. `_fp8_quant`, etc.